pennylane/templates/broadcast.py

```diff
@@ -13,17 +13,45 @@
 # limitations under the License.
 r"""
 Contains the ``broadcast`` template constructor.
-
-To add a new pattern, extend the variables ``OPTIONS``, ``n_parameters`` and ``wire_sequence``, and
-update the list of parameter numbers in the docstring. Optionally add a usage example at the end of the
-``UsageDetails`` directive.
+To add a new pattern:
+* extend the variables ``OPTIONS``, ``n_parameters`` and ``wire_sequence``,
+* update the list in the docstring and add a usage example at the end of the docstring's
+  ``UsageDetails`` section,
+* add tests to parametrizations in :func:`test_templates_broadcast`.
 """
 # pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
 from collections import Iterable
 
 from pennylane.templates.decorator import template
 from pennylane.templates.utils import _check_wires, _check_type, _get_shape, _check_is_in_options
 
+# helpers to define pattern wire sequences
+
+
+def wires_ring(wires):
+    """wire sequence for ring pattern"""
+    ring = list(wires) + list(wires[0:1])
+    sequence = [[ring[i], ring[i + 1]] for i in range(len(wires))]
+    return sequence
+
+
+def wires_pyramid(wires):
```

Comment:
Since this seems to be creating a slightly more complicated structure (could be done for each actually), it would be worth double-checking the correct result by adding 1 or 2 unit tests where:
* `len(wires)` is even
* `len(wires)` is odd

and in each case, `wires` could be something funkier (yet not containing any identical elements).

Reply:
That's a great idea!

Reply:
It actually uncovered a problem with an edge case. Thanks so much!